.github/workflows/main_ci.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Main CI
+on:
+  pull_request:
+    branches: [ main ]
+  merge_group:
+    branches: [ main ]
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}
+  cancel-in-progress: true
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Azure/pipelines@v1.2
+        with:
+          azure-devops-project-url: 'https://dev.azure.com/qiskit-ci/qiskit-terra'
+          azure-pipeline-name: 'Qiskit.qiskit-terra'
+          azure-devops-token: '${{ secrets.AZURE_DEVOPS_TOKEN }}'
```

Comment:
I guess the question is going to be "what `Build.Reason` comes out of this?" for whether the templated pipeline does the right thing.  Looking at the action variables, though, we can probably throw something together with `azure-pipeline-variables` to trigger on if we need.

Reply:
Yeah, reading through the github repo for the action we'll basically have to set variables on the input based on the job type in the gha context if we want to run different subsets of the job for this (to match a PR).